docs/design-documents/docker_management/README.md

```diff
@@ -0,0 +1,137 @@
+
+**fixed tags**
+Using latest tags, user can retrieve an updated version of Docker image. Troubleshooting could become tricky if Docker images are updated and new images are retrieved in the middle of troubleshooting session. Also, some user could decide not to get updated version and would want to stick to a fixed version.
+
+For reasons stated above, all the Docker images are suffixed with a fixed tag. These tags are only applied to one image and never reused. Tags are of the format `mbed-os-<version>-<date>`, example `mbed-os-6.14.0-2021.06.19T04.43.51`, `main-2021.06.19T04.43.51` etc
+
+### Dockerfile versioning
+The mbed-os-env image is created with Dockerfile stored in this repository itself. This provides easy versioning of Dockerfile as it will follows the same versioning strategy of mbed-os. 
+
+### Docker image versioning
+
+The picture below illustrates a typical situation where mbed-os accepts changes to master branch on daily basis and makes releases on regular basis.
+
+![Docker Versioning](./diagrams/docker-versioning.png)
+
+**On Day-X** There are some changes for [active updates](#Type-of-Docker-image-updates) (for example Dockerfile has been changed), so these are the Docker images created or updated
+
+* mbed-os-env:master-latest 
+* mbed-os-env:<master-day-x> - This is a fixed tag
+
+:information_source: master-latest tag is checked for [passive updates](#Type-of-Docker-image-updates) as scheduled in the workflow.
+
+**On Day-X+1** Though there are commits to mbed-os source repository. These do not involve changes to Dockerfile or dependencies like requirements.txt. Hence, no Docker image is created at the time of merging the commit.
+
+**On Day-X+2** A new mbed-os release `mbed-os-6.14.0` is created. This creates an image with following Docker tag.
+
+* mbed-os-env:mbed-os-6-latest - This Docker tag could be used to get a compatible image for latest mbed-os-6 release.
+* mbed-os-env:mbed-os-6.14-latest - This Docker tag  could be used to work with updated mbed-os-6.14 release. This image is passively updated till next mbed-os release on the branch (typically till mbed-os-6.15.0).
+* mbed-os-env:mbed-os-6.14-day-x - A fixed Docker tag.
+
+:information_source: mbed-os-6.14-latest, and mbed-os-6-latest will be passively updated as scheduled in th workflow.
+
+**On Day X+10** Another new release mbed-os-6.15.0 is made. This creates an image with following Docker tag.
+
+* mbed-os-6-latest
+* mbed-os-6.15-latest
+* mbed-os-6.15-day-x
+
+:information_source: From this point, mbed-os-6.14-latest will no longer be passively updated. From this point, only mbed-os-6.15-latest is selected for passive update. Only last release version is passively updated on a branch. 
```

Comment:
You probably want to use fixed-width for tag names here and other places where this was missed.